src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -213,68 +199,57 @@ public IEnumerable<ManifestVersionUpdate> CalculateManifestRollbacks(string roll
             var currentManifestIds = GetInstalledManifestIds();
             var manifestRollbacks = ParseRollbackDefinitionFile(rollbackDefinitionFilePath);
 
-            var unrecognizedManifestIds = manifestRollbacks.Where(rollbackManifest => !currentManifestIds.Contains(rollbackManifest.Item1));
+            var unrecognizedManifestIds = manifestRollbacks.Where(rollbackManifest => !currentManifestIds.Contains(rollbackManifest.Id));
             if (unrecognizedManifestIds.Any())
             {
                 _reporter.WriteLine(string.Format(LocalizableStrings.RollbackDefinitionContainsExtraneousManifestIds, rollbackDefinitionFilePath, string.Join(" ", unrecognizedManifestIds)).Yellow());
-                manifestRollbacks = manifestRollbacks.Where(rollbackManifest => currentManifestIds.Contains(rollbackManifest.Item1));
+                manifestRollbacks = manifestRollbacks.Where(rollbackManifest => currentManifestIds.Contains(rollbackManifest.Id));
             }
 
-            var manifestUpdates = manifestRollbacks
-                .Select(manifest =>
-                {
-                    var installedManifestInfo = GetInstalledManifestVersion(manifest.id);
-                    return new ManifestVersionUpdate(manifest.id, installedManifestInfo.manifestVersion, installedManifestInfo.sdkFeatureBand.ToString(),
-                        manifest.version, manifest.featureBand.ToString());
-                });
+            var manifestUpdates = manifestRollbacks.Select(manifest =>
+            {
+                var (id, (version, band)) = manifest;
+                var (installedVersion, installedBand) = GetInstalledManifestVersion(id);
+                return new ManifestVersionUpdate(id, installedVersion, installedBand.ToString(), version, band.ToString());
+            });
 
             return manifestUpdates;
         }
 
-
         public async Task<IEnumerable<WorkloadDownload>> GetManifestPackageDownloadsAsync(bool includePreviews, SdkFeatureBand providedSdkFeatureBand, SdkFeatureBand installedSdkFeatureBand)
```

Comment:
This method should be checked side-by-side. But the intent is not to cause any functional changes that would affect the operation of the method.